TripAnalysisUI/src/main/java/com/drivequant/drivekit/tripanalysis/workinghours/view/WorkingHoursDayCard.kt

```diff
@@ -0,0 +1,195 @@
+package com.drivequant.drivekit.tripanalysis.workinghours.view
+
+import android.animation.Animator
+import android.animation.AnimatorListenerAdapter
+import android.content.Context
+import android.content.res.ColorStateList
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import androidx.cardview.widget.CardView
+import androidx.core.graphics.drawable.DrawableCompat
+import androidx.core.widget.CompoundButtonCompat
+import com.drivekit.tripanalysis.ui.R
+import com.drivequant.drivekit.common.ui.DriveKitUI
+import com.drivequant.drivekit.common.ui.utils.DKDatePattern
+import com.drivequant.drivekit.tripanalysis.service.workinghours.DKWorkingHoursDayConfiguration
+import com.drivequant.drivekit.tripanalysis.workinghours.toDay
+import com.google.android.material.slider.LabelFormatter
+import com.google.android.material.slider.RangeSlider
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.*
+
+internal class WorkingHoursDayCard : FrameLayout {
+
+    private lateinit var day: DKWorkingHoursDayConfiguration
+
+    private lateinit var cardView: CardView
+    private lateinit var sliderContainer: RelativeLayout
+    private lateinit var rangeSlider: RangeSlider
+    private lateinit var checkbox: CheckBox
+    private lateinit var labelDay: TextView
+    private lateinit var labelMin: TextView
+    private lateinit var labelMax: TextView
+
+    private var listener: WorkingHoursDayListener? = null
+
+    constructor(context: Context, day: DKWorkingHoursDayConfiguration) : super(context) {
+        init(day)
+    }
+
+    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
+        init(day)
+    }
+
+    private fun init(dayConfig: DKWorkingHoursDayConfiguration) {
+        this.day = dayConfig
+        val view = View.inflate(context, R.layout.dk_layout_working_hours_day_item, null)
+        cardView = view.findViewById(R.id.cardview)
+        sliderContainer = view.findViewById(R.id.rangeslider_container)
+        rangeSlider = view.findViewById(R.id.range_slider)
+        checkbox = view.findViewById(R.id.checkbox_select_day)
+        labelDay = view.findViewById(R.id.card_view_day_textview)
+        labelMin = view.findViewById(R.id.textMin)
+        labelMax = view.findViewById(R.id.textMax)
+
+        checkbox.apply {
+            setOnCheckedChangeListener { _, isChecked ->
+                listener?.onDayChecked(isChecked)
+                manageCheckboxStyle(isChecked)
+                manageSliderVisibility(isChecked)
+            }
+            isChecked = !dayConfig.entireDayOff
+        }
+
+        labelDay.apply {
+            val cal = Calendar.getInstance(Locale.getDefault())
+            cal[Calendar.DAY_OF_WEEK] = dayConfig.day.toDay()
+            text = SimpleDateFormat("E", Locale.getDefault()).format(cal.time)
```

Comment:
Utiliser `DKDatePattern.YEAR_ONLY.getPattern()`

Reply:
Hello, en effet le pattern `E` est bon, mais cela ne correspond pas à la définition de "YEAR ONLY". Avec `"E"`, on récupère le "day name in week" c'est pourquoi je n'ai pas utilisé `DKDatePattern` ici. Nous corrigerons le bug dans un ticket séparé